replica/distributed_loader.cc

```diff
@@ -452,12 +458,12 @@ distributed_loader::process_upload_dir(distributed<replica::database>& db, distr
         };
         process_sstable_dir(directory, flags).get();
 
-        std::vector<std::unique_ptr<sstable_generation_generator>> gen;
+        std::vector<std::unique_ptr<sstable_generation_generator<true>>> gen;
```

Comment:
Please use a tagged type, looking at the code here it is impossible to guess what this `true` means.
You can use something like `use is_shared = bool_class<struct is_shared_tag>;`, then, you will have `is_shared::yes` here, which is much more readable.

Reply:
i am afraid that we cannot use an instance of `bool_class` as the template parameter. as it is not a "structural type", see https://en.cppreference.com/w/cpp/language/template_parameters -- it has non-static member variable. so i just added a 

```c++
struct is_shared {
     using yes = std::true_type;
     using no = std::false_type;
};
```

just for better readability instead of enforcing using yet another type by defining a new type using `std::integral_constant` or having a template class inheriting from `bool_constant<B>`. both approaches look like overkill.